### DISCUSSION TITLE
Compile-time errors in generated code

### DISCUSSION THREAD

**Discussion by ModernRonin** [canton7/PropertyChanged.SourceGenerator#19] [Q&A] [answered]
Hi,
after adding the PC.SG package and converting a (very simple) viewmodel to use it, I get loads of compile-time errors in the generated code. 

Any idea what could be the issue here?

My csproj looks like this:

```xml
<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net6.0-windows</TargetFramework>
		<UseWPF>true</UseWPF>

		<RootNamespace>Viewer</RootNamespace>
	</PropertyGroup>

	<ItemGroup>
		<PackageReference Include="ModernWpfUI" Version="0.9.6" />
		<PackageReference Include="PropertyChanged.SourceGenerator" Version="1.0.4">
			<PrivateAssets>all</PrivateAssets>
			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
		</PackageReference>
		<PackageReference Include="Stylet" Version="1.3.6.0" />
	</ItemGroup>
</Project>
```

My viewmodel looks like:

```csharp
using PropertyChanged.SourceGenerator;

namespace Viewer.Pages;

public partial class HexConfigurationViewModel
{
    [Notify] int _ringCount = 5;
}
```

and the errors I get are:

```text
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(38,20,38,35): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for 'NotifyAttribute'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(72,20,72,39): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for 'AlsoNotifyAttribute'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(86,20,86,38): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for 'DependsOnAttribute'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(100,20,100,38): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for 'IsChangedAttribute'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(10,19,10,25): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for 'Getter'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(23,19,23,25): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for 'Setter'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\EventArgsCache.cs(6,27,6,41): error CS0101: The namespace 'PropertyChanged.SourceGenerator.Internal' already contains a definition for 'EventArgsCache'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(70,6,70,35): error CS0579: Duplicate 'global::System.AttributeUsage' attribute
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(36,6,36,35): error CS0579: Duplicate 'global::System.AttributeUsage' attribute
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\HexConfigurationViewModel.cs(8,80,8,95): error CS0102: The type 'HexConfigurationViewModel' already contains a definition for 'PropertyChanged'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\HexConfigurationViewModel.cs(9,20,9,29): error CS0102: The type 'HexConfigurationViewModel' already contains a definition for 'RingCount'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\HexConfigurationViewModel.cs(25,32,25,49): error CS0111: Type 'HexConfigurationViewModel' already defines a member called 'OnPropertyChanged' with the same parameter types
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(84,6,84,35): error CS0579: Duplicate 'global::System.AttributeUsage' attribute
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(98,6,98,35): error CS0579: Duplicate 'global::System.AttributeUsage' attribute
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(43,16,43,31): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the same parameter types
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(51,16,51,31): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the same parameter types
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(58,16,58,31): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the same parameter types
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(64,16,64,31): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the same parameter types
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(78,16,78,35): error CS0111: Type 'AlsoNotifyAttribute' already defines a member called 'AlsoNotifyAttribute' with the same parameter types
1>C:\Projects\Github\HexExperiments\Viewer\Pages\HexConfigurationViewModel.cs(7,6,7,12): error CS0121: The call is ambiguous between the following methods or properties: 'NotifyAttribute.NotifyAttribute()' and 'NotifyAttribute.NotifyAttribute()'
1>C:\Projects\Github\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator\Attributes.cs(92,16,92,34): error CS0111: Type 'DependsOnAttribute' already defines a member called 'DependsOnAttribute' with the same parameter types
```

I also tried adding the `IncludePackageReferencesDuringMarkupCompilation` directive to the project file (although, AFAIU, the issue necessitating this is fixed for net6), this had no effect.

**Comment by canton7** [comment, marked as answer]
Interesting. I can't replicate this using that csproj and ViewModel.

It's really weird that you're seeing generated files in `PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator`. The generated files should be kept safely in the `obj` folder (there's an option for moving them out of there, but your csproj doesn't have that enabled). Did you move the generated files there manually? Can you delete them and see whether that fixes the issue?

**Reply by ModernRonin** [reply]
I didn't move anything. (I must also warn you: my experience with source generators is basically zero currently). I'll try deleting them and report back.

**Reply by ModernRonin** [reply]
interestingly, the files are NOT at that location - there is not even a folder PropertyChanged.SourceGenerator inside Viewer.

**Reply by ModernRonin** [reply]
so I deleted bin and obj, closed VS and built from the command-line, just to be sure, and I get the same errors.

**Reply by canton7** [reply]
That's exciting. Can you try restarting VS and deleting your obj folder? Can you share a full project which reproduces the issue which I can run?

**Reply by canton7** [reply]
(This is almost certainly a tooling issue of some sort btw, but I'm not sure what sort of tooling issue it might be)

**Reply by ModernRonin** [reply]
I just made the repo public - it was just private because I don't think it's interesting to anyone, just playing around in my last vacation day ;P.

The repo and relevant branch is 
[use-propertychanged-sourcegen](https://github.com/ModernRonin/HexExperiments/tree/use-propertychanged-sourcegen).

as for restarting, yeah, I did that already, to no avail.

**Reply by canton7** [reply]
Huh, that builds just fine here:

```
> dotnet build
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Projects\Temp\HexExperiments\Viewer\Viewer.csproj (in 356 ms).
  Viewer -> C:\Projects\Temp\HexExperiments\Viewer\bin\Debug\net6.0-windows\Viewer.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:04.96

> dotnet --version
6.0.300

> dotnet --list-sdks
2.0.0 [C:\Program Files\dotnet\sdk]
6.0.300 [C:\Program Files\dotnet\sdk]
```

Builds fine from VS 17.2.1 as well.

**Reply by ModernRonin** [reply]
(might also be some stupid mistake on my part, I haven't done any WPF for probably 10 years or so, always just backend, and now I'm just fiddling for my own fun).

because you mention tooling, dotnet --version outputs 6.0.302, and my VS is 17.2.6

**Reply by ModernRonin** [reply]
aha, so we got different net sdks and VS versions

**Reply by canton7** [reply]
Let me update VS, see if that changes anything

**Reply by canton7** [reply]
Can you just try a clean clone of that repo, checkout ac7893e4?

**Reply by canton7** [reply]
(Going to grab some food, back in a bit)

**Reply by ModernRonin** [reply]
did that (clone anew to my ramdisk, checkout the commit, dotnet build), same result:

> C:\Projects\Github\HexExperiments [use-propertychanged-sourcegen ≡]> r:
R:\> git clone https://github.com/ModernRonin/HexExperiments.git
Cloning into 'HexExperiments'...
remote: Enumerating objects: 85, done.
remote: Counting objects: 100% (85/85), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 85 (delta 45), reused 69 (delta 29), pack-reused 0Receiving objects:  31% (27/85)
Receiving objects: 100% (85/85), 14.84 KiB | 2.97 MiB/s, done.
Resolving deltas: 100% (45/45), done.
R:\> cd .\HexExperiments\
R:\HexExperiments [(master)]> git checkout ac7893e4
fatal: detected dubious ownership in repository at 'R:/HexExperiments'
To add an exception for this directory, call:

        git config --global --add safe.directory R:/HexExperiments

Set the environment variable GIT_TEST_DEBUG_UNSAFE_DIRECTORIES=true and run
again for more information.
R:\HexExperiments [(master)]> git config --global --add safe.directory R:/HexExperiments
R:\HexExperiments [master ≡]> git checkout ac7893e4
Note: switching to 'ac7893e4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at ac7893e tried to integrate PropertyChanged.SourceGenerator, fails with errors, waiting for answer at https://github.com/canton7/PropertyChanged.SourceGenerator/discussions/19
R:\HexExperiments [(ac7893e...)]> dotnet build
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored R:\HexExperiments\Viewer\Viewer.csproj (in 213 ms).
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(38,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'NotifyAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(72,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'AlsoNotifyAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(86,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'DependsOnAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(100,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
 'IsChangedAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(10,19): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'Getter' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(23,19): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'Setter' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\HexConfigurationViewModel.cs(8,80): error CS0102: The type 'HexConfigurationViewModel' already contains a definition f
or 'PropertyChanged' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\HexConfigurationViewModel.cs(9,20): error CS0102: The type 'HexConfigurationViewModel' already contains a definition f
or 'RingCount' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\HexConfigurationViewModel.cs(25,32): error CS0111: Type 'HexConfigurationViewModel' already defines a member called 'O
nPropertyChanged' with the same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\EventArgsCache.cs(6,27): error CS0101: The namespace 'PropertyChanged.SourceGenerator.Internal' already contains a def
inition for 'EventArgsCache' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(36,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(70,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(84,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(98,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\Pages\HexConfigurationViewModel.cs(7,6): error CS0121: The call is ambiguous between the follo
wing methods or properties: 'NotifyAttribute.NotifyAttribute()' and 'NotifyAttribute.NotifyAttribute()' [R:\HexExperime
nts\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(43,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(51,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(58,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(64,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(92,16): error CS0111: Type 'DependsOnAttribute' already defines a member called 'DependsOnAttribute' wit
h the same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(78,16): error CS0111: Type 'AlsoNotifyAttribute' already defines a member called 'AlsoNotifyAttribute' w
ith the same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]

Build FAILED.

R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(38,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'NotifyAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(72,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'AlsoNotifyAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(86,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'DependsOnAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(100,20): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
 'IsChangedAttribute' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(10,19): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'Getter' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(23,19): error CS0101: The namespace 'PropertyChanged.SourceGenerator' already contains a definition for
'Setter' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\HexConfigurationViewModel.cs(8,80): error CS0102: The type 'HexConfigurationViewModel' already contains a definition f
or 'PropertyChanged' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\HexConfigurationViewModel.cs(9,20): error CS0102: The type 'HexConfigurationViewModel' already contains a definition f
or 'RingCount' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\HexConfigurationViewModel.cs(25,32): error CS0111: Type 'HexConfigurationViewModel' already defines a member called 'O
nPropertyChanged' with the same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\EventArgsCache.cs(6,27): error CS0101: The namespace 'PropertyChanged.SourceGenerator.Internal' already contains a def
inition for 'EventArgsCache' [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(36,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(70,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(84,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(98,6): error CS0579: Duplicate 'global::System.AttributeUsage' attribute [R:\HexExperiments\Viewer\Viewe
r_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\Pages\HexConfigurationViewModel.cs(7,6): error CS0121: The call is ambiguous between the follo
wing methods or properties: 'NotifyAttribute.NotifyAttribute()' and 'NotifyAttribute.NotifyAttribute()' [R:\HexExperime
nts\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(43,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(51,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(58,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(64,16): error CS0111: Type 'NotifyAttribute' already defines a member called 'NotifyAttribute' with the
same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(92,16): error CS0111: Type 'DependsOnAttribute' already defines a member called 'DependsOnAttribute' wit
h the same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
R:\HexExperiments\Viewer\PropertyChanged.SourceGenerator\PropertyChanged.SourceGenerator.PropertyChangedSourceGenerator
\Attributes.cs(78,16): error CS0111: Type 'AlsoNotifyAttribute' already defines a member called 'AlsoNotifyAttribute' w
ith the same parameter types [R:\HexExperiments\Viewer\Viewer_40i3lxko_wpftmp.csproj]
    0 Warning(s)
    21 Error(s)

Time Elapsed 00:00:01.62
R:\HexExperiments [(ac7893e...)]> dotnet --version
6.0.302
R:\HexExperiments [(ac7893e...)]> dotnet --list-sdks
3.1.402 [C:\Program Files\dotnet\sdk]
3.1.421 [C:\Program Files\dotnet\sdk]
5.0.103 [C:\Program Files\dotnet\sdk]
5.0.104 [C:\Program Files\dotnet\sdk]
5.0.214 [C:\Program Files\dotnet\sdk]
5.0.408 [C:\Program Files\dotnet\sdk]
6.0.203 [C:\Program Files\dotnet\sdk]
6.0.302 [C:\Program Files\dotnet\sdk]
R:\HexExperiments [(ac7893e...)]>

**Reply by ModernRonin** [reply]
so far, the only difference I can see is that you build with 6.0.300 and it works fine, while I build with 6.0.302 and it fails.

**Reply by ModernRonin** [reply]
and there's the wrong path in the error message although the files are NOT there. actually, if I search for Attributes.cs inside the repo, I don't find it at all. so maybe the compiler deletes it after building?

**Reply by canton7** [reply]
Aha! Upgraded to 6.0.302 / VS 17.2.6, and I can reproduce this.

**Reply by canton7** [reply]
Interestingly, my little test project is fine.

**Reply by ModernRonin** [reply]
will be interesting what's the difference between the two (your test project and mine)

**Reply by ModernRonin** [reply]
btw, somewhat unrelated, but let me say you are one of my personal heroes for creating Stylet - when I noticed that Caliburn.Micro is out of commission, I was temporarily very sad until I found Stylet - and I *love* that you removed all the magic stuff Caliburn used to have, that just tended to create trouble sooner or later when the projects got more complex.

**Reply by canton7** [reply]
I can make it build by deleting parts of your project. I haven't yet managed to narrow down exactly what causes it though -- every time I think I've got it I restore everything and re-make the latest change, and it fails to build again.

**Reply by canton7** [reply]
This looks similar: https://github.com/dotnet/wpf/issues/6792

**Reply by canton7** [reply]
And that [workaround](https://github.com/dotnet/wpf/issues/6792#issuecomment-1183633402) fixes it. Yay, WPF bug.

**Reply by ModernRonin** [reply]
cool, thanks a LOT!

**Reply by ModernRonin** [reply]
That workaround is nice also in being fairly general

**Reply by canton7** [reply]
Glad that worked! Annoying that you just happened to start using SGs with WPF at the precise version where they were buggy... Hopefully everything is smooth from now on!
